### ISSUE TITLE
IfcSectionedSurface, IfcSectionedSolidHorizontal and IfcOpenCrossProfileDef Suggested Improvements

### ISSUE THREAD

**Issue by jmirtsch** [bSI-InfraRoom/IFC-Specification#92]
In reviewing the IfcSectionedSurface unit test https://github.com/bSI-InfraRoom/IFC-infra-unit-test/pull/106

I've identified some inconsistencies and redundancies, and would like to raise the following issues.

1. Unexpected inconsistency between IfcSectionedSolidHorizontal and IfcSectionedSurface.
  IfcSectionedSolidHorizontal attribute CrossSectionPositions : LIST [2:?] OF IfcAxis2PlacementLinear and I'd suggest IfcSectionedSurface should also adopt this.
2. IfcSectionedSolidHorizontal attribute FixedAxisVertical is redundant, as this can be achieve by nominating (0,0,1) on the Axis attribute of IfcAxis2PlacementLinear
3. At present, IfcOpenCrossProfileDef curve seems to have an implicit starting point at origin (0,0).  Rotating the position plane is problematic if you have to use IfcPointByDistanceExpression attribute OffsetVertical to offset the curve in a vertical sense as this will occur around the RefDirection of the position which won't be consistent with the directrix.

Schema suggestions
1.  Change IfcSectionedSurface CrossSectionPositions attribute type to IfcAxis2PlacementLinear.  
2. Remove FixedAxisVertical : IfcBoolean;from IfcSectionedSolidHorizontal and IfcSectionedSurface as the position can control this.
3. Add a start point attribute to the IfcOpenCrossProfileDef so that the position does not require to be offset laterally or vertically from the directrix.

I'll submit a pull request related to these suggestions, also incorporating some documentation adjustments/corrections.

**Comment by javierNadal** [comment]
I don't think that "Add a start point attribute to the IfcOpenCrossProfileDef so that the position does not require to be offset laterally or vertically from the directrix." is a good idea. 
Lateral and vertical offset is good for more complex situations. With only one alignment (horizontal + vertical) you can make multiple IfcSectionedSurfaces for several road layers, each one with his own offset.

**Comment by SergejMuhic** [comment]
Indeed @javierNadal . This can be restricted for road cases in the road exchange in whatever MVD would be used. But maybe a general base entity is needed for exchanges other than road.

**Comment by jmirtsch** [comment]
It's not easy to get a good image that depicts the issue I'm trying to convey with the suggestion for a start point of an IfcOpenCrossProfileDef.
Maybe I misunderstand the requirement, but I would expect that the basis curve (alignment) would define the extent of the sectioned surface with forced vertical profile.  If an IfcAxis2PlacementLinear is used to position the profile, then when forcing it to a vertical plane using the Axis attribute of the IfcAxis2PlacementLinear, the pivot point can't be offset vertically or the resultant vertical profile will be offset from the basis curve.  If the IfcOpenCrossProfileDef can start at a point away from 0,0 then the pivot point can be constrained to the basis curve (ie offsets are zero).

![210710 sectioned surface undesired offset to basis curve](https://user-images.githubusercontent.com/791757/125162506-9804e380-e1cb-11eb-8f23-e470a099179c.png)

**Comment by TLiebich** [comment]
isn't it a more general question? IfcPointByDistanceExpression defines the vertical offset as being perpendicular to the gradient (local z), whereas it is meant (at least in context of the secioned surface and sectioned solid) to be in global Z. Having the origin of the profile being in 0.,0. would just be the exception, where it does not matter?
